### PR TITLE
Current Club Block Docs & Tests

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request adds some documentation and Cypress tests for the new Current Club Block added in #2345 

### How should this be reviewed?
👀 ([Docs on Gitbooks](https://dosomething.gitbook.io/phoenix-documentation/v/docs%2Fcurrent-club-block-+-tests/development/content-types/current-club-block)).

### Any background context you want to provide?
I had much trouble trying to test the 'lazy auth' flow for unauthenticated users.

[We have](https://github.com/DoSomething/phoenix-next/blob/17165454b0f30abc35a5ef724a22fe7c3a384855/cypress/support/commands.js#L81-L102) a `handleLogin` Cypress command, which _should_ take care of everything after adding the Cypress logic we use in our Redux [lazy auth](https://github.com/DoSomething/phoenix-next/blob/17165454b0f30abc35a5ef724a22fe7c3a384855/resources/assets/middleware/requiresAuthentication.js#L26-L31) to the [Hooks flow](https://github.com/DoSomething/phoenix-next/blob/17165454b0f30abc35a5ef724a22fe7c3a384855/resources/assets/helpers/auth.js#L112-L115). But for some reason, Cypress fails [to find](https://github.com/DoSomething/phoenix-next/blob/17165454b0f30abc35a5ef724a22fe7c3a384855/cypress/support/commands.js#L89) the `data-test="redirect"` in the DOM though it clearly seemed to be in there when I was attempting this.

In any case, I'm not positive how we'd mock the core functionality which is the `updateClubId` mutation running with the preselected and 'flashed' `clubId` in session storage, since I don't think our [Cypress GraphQL Mock](https://github.com/tgriesser/cypress-graphql-mock) is sophisticated enough to allow that? So I left this test as a TODO.

### Relevant tickets

References [Pivotal #174340202](https://www.pivotaltracker.com/story/show/174340202).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [x] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [x] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js [hunk 1]
```diff
@@ -46,17 +46,20 @@ const CurrentClubForm = () => {
 
   // If the mutation was successful, hide the form while the CurrentClubBlock
   // updates to display the user's current club.
-  if (data) {
+  if (isAuthenticated() && (data || flash.clubId)) {
```
Comment (katiecrane) [inline resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js:49]: Okay I see how if this is loading then the user must be authenticated - but when would we have only `data` or `flash.clubId`?

Comment (mendelB) [reply resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js:49]: Yeah this one escaped me on the first run! But if the user's doing lazy auth, there will be a couple of seconds potentially, when they've returned from Northstar and the mutation is gonna be triggered or is triggered and now loading, and we don't want to display the form -- and we'd catch this by seeing that there's a `flash.clubId`!

Comment (mendelB) [reply resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js:49]: Hmm perhaps a cooler UI would be to populate the form with their preselected ClubId and use the `loading` to make it look more like a loading form, but I think this is the quicker bet for now.

File: cypress/integration/current-club-block.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      cy.findByTestId('current-club').within(() => {
+        cy.findByTestId('current-club-name').contains(club.name);
+      });
+    });
+  });
+
+  context('Unauthenticated users', () => {
+    // @TODO: Fix this test to authenticate the user and see if the mutation runs with their selected club ID.
+    /** @test */
+    xit('Can join a club via "lazy authentication"', () => {
```
Comment (katiecrane) [inline cypress/integration/current-club-block.js:73]: Does `xit` mean it doesn't run?

File: resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js (hunk 1 above)
Comment (katiecrane) [reply resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js:49]: Oooooo so before we have `data` back we need to be in a loading state and at that time we'll only have `flash.clubId`?

**Comment by katiecrane** [comment]
>I had much trouble trying to test the 'lazy auth' flow for unauthenticated users.

This test sounds like a doozy! I think we should be sure to flag this as a specific flow to test when we're doing QA since we don't have Cypress support for it

File: resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js (hunk 1 above)
Comment (mendelB) [reply resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js:49]: Exactly! 👌

**Comment by mendelB** [comment]
> > I had much trouble trying to test the 'lazy auth' flow for unauthenticated users.
> 
> This test sounds like a doozy! I think we should be sure to flag this as a specific flow to test when we're doing QA since we don't have Cypress support for it

Oooh good flag -- will do!

File: resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js (hunk 1 above)
Comment (katiecrane) [reply resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js:49]: Okay now I might be confusing myself - but doesn't `data` being truthy imply that we have a result back from the mutation and if that is so, won't this _always_ show the spinner even once we have a result? This also might just be Monday morning brain

Comment (mendelB) [reply resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js:49]: No, you're absolutely right and that's intentional! Once we have a result, we want the spinner to show until the `UserClubQuery` runs again and the `CurrentClubBlock` [swaps](https://github.com/DoSomething/phoenix-next/blob/17165454b0f30abc35a5ef724a22fe7c3a384855/resources/assets/components/blocks/CurrentClubBlock/CurrentClubBlock.js#L48-L70) the current club view in and gets rid of the form!

Comment (katiecrane) [reply resources/assets/components/blocks/CurrentClubBlock/CurrentClubForm.js:49]: AH I was just missing the last step - thanks for walking me through it!!

**Pull request merged**